### PR TITLE
enhancement: add visualization script to annotate elements

### LINKED ISSUE #1604: bug: synchronize the default layout model
**Describe the bug**
Currently, `unstructured` and `unstructured-inference` use different default layout models, so there are many differences in the elements extracted by the two libraries.
- unstructured: [detectron2_onnx](https://github.com/Unstructured-IO/unstructured/blob/main/unstructured/partition/pdf.py#L332-L335) (old model)
- unstructured-inference: [yolox_quantized](https://github.com/Unstructured-IO/unstructured-inference/blob/main/unstructured_inference/models/base.py#L26) (updated in the `unstructured-inference` [PR #201](https://github.com/Unstructured-IO/unstructured-inference/pull/201/files#diff-b69ebd085260ed1336b50dce46c6246ff3a2ab36367e7086eaf11716f2de4552R26))

**To Reproduce**
PDF: [references.pdf](https://github.com/Unstructured-IO/unstructured/files/12778270/references.pdf)
- `unstructured`
```
    elements = partition_pdf(filename, strategy="hi_res")
    print(len(elements))
```
- `unstructured-inference`
```
    layout = process_file_with_model(filename="references.pdf", model_name=None)
    print(len(layout.pages[0].elements))
```
**Screenshots**
- Elements extracted with the `unstructured` Library
hi_res-1

- Elements extracted with the `unstructured-inference` Library
references_1_final

**Expected behavior**
The elements extracted by the two libraries shouldn't be too different.

**Environment Info**
```
unstructured             0.10.18
unstructured-inference   0.6.6
```
**Additional context**
This issue is related to issue #1602.

### PR DESCRIPTION
**Pull request by christinestraub**
This PR was initially created to close #1604 (Synchronizing the default layout model), but since it was already resolved in PR [#1607](https://github.com/Unstructured-IO/unstructured/pull/1607), this PR now only adds the visualization script used to investigate the issue.
### Summary
- add python script to annotate elements

PDF: [references.pdf](https://github.com/Unstructured-IO/unstructured/files/12778270/references.pdf)

### Evaluation
```
PYTHONPATH=. python examples/layout-analysis/visualization.py references.pdf hi_res
```

### CODE REVIEW

File: unstructured/partition/pdf.py [hunk 1]
```diff
@@ -332,7 +332,7 @@ def _partition_pdf_or_image_local(
     model_name = (
         model_name
         if model_name
-        else os.environ.get("UNSTRUCTURED_HI_RES_MODEL_NAME", "detectron2_onnx")
+        else os.environ.get("UNSTRUCTURED_HI_RES_MODEL_NAME", "yolox_quantized")
```
Comment (cragwolfe) [inline unstructured/partition/pdf.py:335]: It was intentional to stick with the old model to isolate differences in ingest tests diff between the library change itself, and then the model change.

Related PR: https://github.com/Unstructured-IO/unstructured/pull/1607

CC @badGarnet 

The examples/layout-analysis/visualization.py script looks like a useful addition!

Comment (christinestraub) [reply unstructured/partition/pdf.py:335]: Got it, Thanks for letting me know! I'll hold this PR until [PR #1607](https://github.com/Unstructured-IO/unstructured/pull/1607) is merged.

File: examples/layout-analysis/visualization.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+
+    for idx, (img, coords_per_page) in enumerate(zip(images, elements_coordinates)):
+        for coordinate in coords_per_page:
+            points = coordinate.points
+            x1, y1 = points[0]
+            x2, y2 = points[2]
+            rect = Rectangle(x1, y1, x2, y2)
+            img = draw_bbox(img, rect, color="red")
+
+        output_image_path = os.path.join(output_dir, f"{strategy}-{idx + 1}.jpg")
```
Comment (cragwolfe) [inline examples/layout-analysis/visualization.py:52]: please print the path here for convenience

Comment (christinestraub) [reply examples/layout-analysis/visualization.py:52]: Sounds good!

**Pull request merged**